docs/addsegment-guide.md

```diff
@@ -0,0 +1,78 @@
+<!--
+    Licensed to the Apache Software Foundation (ASF) under one or more 
+    contributor license agreements.  See the NOTICE file distributed with
+    this work for additional information regarding copyright ownership. 
+    The ASF licenses this file to you under the Apache License, Version 2.0
+    (the "License"); you may not use this file except in compliance with 
+    the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+    
+    Unless required by applicable law or agreed to in writing, software 
+    distributed under the License is distributed on an "AS IS" BASIS, 
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and 
+    limitations under the License.
+-->
+
+# Heterogeneous format segments in carbondata
+
+###Background
+In the industry, many users already adopted to data with different formats like ORC, Parquet, JSON, CSV etc.,  
+If users want to migrate to Carbondata for better performance or for better features then there is no direct way. 
+All the existing data needs to be converted to Carbondata to migrate.  
+This solution works out if the existing data is less, what if the existing data is more?   
+This process of converting is very time consuming and error-prone.
```

Comment:
Line 25 can be changed to "Heterogeneous format segments aims to solve this problem by avoiding data conversion"

Reply:
done